parsec/core/types/manifest.py

```diff
@@ -547,11 +554,17 @@ def _restore_local_confinement_points(
         # before applying a new filter that filtered those entries from the remote manifest
         if not other.local_confinement_points and not self.remote_confinement_points:
             return self
+        # Create a set for fast lookup in order to make sure no entry gets duplicated.
+        # This might happen when a synchronized entry is renamed to a confined name locally.
+        self_entry_ids = set(self.children.values())
         previously_local_confinement_points = {
             name: entry_id
             for name, entry_id in other.children.items()
-            if entry_id in other.local_confinement_points
-            or entry_id in self.remote_confinement_points
+            if entry_id not in self_entry_ids
```

Comment:
So if an entry has been renamed in local to a confined name, we revert the rename here ? (even if the remote change that triggered the sync is not related to the entry, typically a new entry added to the folder manifest)

Reply:
The rename is not reverted here, this method is mostly used to create `local_from_remote`, i.e a local manifest that corresponds to the remote once the local context is applied (but without merging with the content of the local manifest). Then this `local_from_remote` is merged with the `local` manifest and potential conflicts get resolved.

In the case we're interested in, this should happen:
- `local_from_remote` contains some changes 
- `local_manifest` contains a rename to a confinement point
- if the changes in `local_from_remote` do not include a rename of the sensitive entry, the merge gets resolved without conflict, and the new manifest includes both the remote changes and the rename to confinement.
- if `local_from_remote` includes a rename of the sensitive entry, the merge finds a conflict that gets resolved like other similar conflicts (in this case renaming the entry with a special suffix indicating what happened)

Reply:
👍